### PR TITLE
Add the ability to specify HTTP headers for CLI commands (to support Auth proxies)

### PR DESCRIPTION
**Pull request by nchelluri**
# Description of the PR

Fixes https://github.com/guacsec/guac/issues/1810

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x] All new changes are covered by tests
- [x] If GraphQL schema is changed, `make generate` has been run
- [x] If OpenAPI spec is changed, `make generate` has been run
- [x] If `collectsub` protobuf has been changed, `make proto` has been run
- [x] All CI checks are passing (tests and formatting)
- [x] All dependent PRs have already been merged

### CODE REVIEW

File: cmd/guacgql/cmd/server.go [hunk 1]
```diff
@@ -81,9 +83,8 @@ func startServer(cmd *cobra.Command) {
 		logger.Errorf("unable to initialize graphql server: %v", err)
 		os.Exit(1)
 	}
-	// Setup Prometheus metrics handler
-	metric, err := cli.SetupPrometheus(ctx, logger, "guacgql")
```
Comment (mihaimaruseac) [inline cmd/guacgql/cmd/server.go:85]: Why did we move this function?

**Review by mihaimaruseac — COMMENTED** [review]
Thank you

File: cmd/guacgql/cmd/server.go (hunk 1 above)
Comment (nchelluri) [reply cmd/guacgql/cmd/server.go:85]: I was looking at that function a few PRs ago and when I revisited the `"github.com/guacsec/guac/pkg/cli` package in the course of this header work, I saw that there was no other caller of the function and felt it was misplaced in `cli/init.go`. So I decided to move it to where it was used and to make it private when I did. It is not relevant to the CLI command HTTP header functionality.

**Review by pxp928 — APPROVED** [review]
LGTM

File: cmd/guacgql/cmd/server.go (hunk 1 above)
Comment (mihaimaruseac) [reply cmd/guacgql/cmd/server.go:85]: Thank you for the explanation!

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**